File: problems/linked_lists/longestStreak.js
const longestStreak = head => {
  let count = 0;
  let longest = 0; 
  let prevVal = null 
  while(head) {
  
    if(head.val === prevVal) {
      count++
    } else {

      count = 1
    }
    longest = Math.max(longest, count)
    prevVal = head.val 
    head = head.next 
    
  }
 
  
  return longest 
}

File: python_Structy/longest_streak.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.next = None

def longest_streak(head):
  if head is None: return 0
  count = 0
  maxCount = 0
  currVal = head.val
  while(head):
    if head.val == currVal: count += 1 
    else: 
      maxCount = max(maxCount, count)
      count = 1 
      currVal = head.val 
    head = head.next 

  return max(maxCount, count)
